src/mpid/ch4/src/mpidig_part.h

```diff
@@ -9,54 +9,38 @@
 #include "ch4_impl.h"
 #include "mpidig_part_utils.h"
 
-void MPIDIG_precv_matched(MPIR_Request * part_req);
 int MPIDIG_mpi_psend_init(const void *buf, int partitions, MPI_Aint count,
                           MPI_Datatype datatype, int dest, int tag,
                           MPIR_Comm * comm, MPIR_Info * info, MPIR_Request ** request);
 int MPIDIG_mpi_precv_init(void *buf, int partitions, MPI_Aint count,
                           MPI_Datatype datatype, int source, int tag,
                           MPIR_Comm * comm, MPIR_Info * info, MPIR_Request ** request);
 
-MPL_STATIC_INLINE_PREFIX int MPIDIG_part_start(MPIR_Request * request)
-{
-    int mpi_errno = MPI_SUCCESS;
-    MPIR_FUNC_ENTER;
-
-    MPID_THREAD_CS_ENTER(VCI, MPIDI_VCI(0).lock);
-
-    /* Indicate data transfer starts.
-     * Decrease when am request completes on sender (via completion_notification),
-     * or received data transfer AM on receiver. */
-    MPIR_cc_set(request->cc_ptr, 1);
-
-    /* No need to increase refcnt for comm and datatype objects,
-     * because it is erroneous to free an active partitioned req if it is not complete.*/
-
-    if (request->kind == MPIR_REQUEST_KIND__PART_RECV && MPIDIG_PART_REQUEST(request, peer_req_ptr)) {
-        mpi_errno = MPIDIG_part_issue_cts(request);
-    }
-
-    MPIR_Part_request_activate(request);
-
-    MPID_THREAD_CS_EXIT(VCI, MPIDI_VCI(0).lock);
-    MPIR_FUNC_EXIT;
-    return mpi_errno;
-}
```

Comment:
moving this function to mpidig_part.c is probably worth separating into another commit. then make modifications to it.